### ISSUE TITLE
user_ref in Structured Message

### ISSUE THREAD

**Issue by zewlak** [pimax/fb-messenger-php#97]
Is it possible to send Structured Message to recipient:user_ref instead of recipient:id? 
There is no way to pass user_ref param to Structured Message. Maybe a new public method setUserRef in Message class would do the job (for now)? It would also cover this problem for another Message types like ImageMessage, QuickReply etc.

What do you think?

**Comment by wittfabian** [comment]
Please provide a link to the documentation.

**Comment by zewlak** [comment]
Sadly it's undocumented that you can send rich messages to user_ref but I tested it and works fine.

Sending to user_ref is described only here in last paragraph (Calling the Send API)
https://developers.facebook.com/docs/messenger-platform/plugin-reference/checkbox-plugin

**Comment by wittfabian** [comment]
recipient with user_ref is only for the first call.

> After you receive the callback event, you can call the Send API to start messaging the user using the user_ref identifier in recipient as shown below. Note that this field is the same as the unique user_ref param used before when the plugin was rendered and in confirming the opt-in.

> If the call to the Send API is successful, the response will contain a recipient_id parameter, which is a stable user ID that you can now use in future API calls.

**Comment by zewlak** [comment]
That's right but response is sent after first user interaction with message sent in first call. I'm sending Message with Buttons in first call to increase response rate. Like below:

![tested](https://image.ibb.co/i6MfVk/github.png)
